### ISSUE TITLE
Forced 'format' when running via the CLI (rather than autodetect)

### ISSUE THREAD

**Issue by andylockran** [stoplightio/spectral#1606]
**User story.**
As a api developer, I can force spectral to use a particular specification format, so that I can override the autodetected format.

**Is your feature request related to a problem?**

I use AWS API Gateway, and they've got a fairly heavily bastardised version of OAS 3.0 in use (as well as using jsonschema draft-4 for their model validation, rather than OAS3.0 validation).

I've generated a jsonschema that articulates some of their non-standard implementation rules:
https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-known-issues.html
but as it describes the document with a standards-compliant OAS header, there's no way currently to force spectral to use a non-standard format to validate a document that appears to be standard.

**Describe the solution you'd like**
A new CLI flag to be able to force the format used to evaluate a document.  Since --format already defines the output format, I'd want the maintainers suggestion on what the flag should be called before I prepare a PR.

**Additional context**

**Comment by andylockran** [comment]
Just for clarification, I think the reason this is a new format - rather than a set of rulesets ontop of the OAS3.0 schema, is due to the mix of OAS and jsonschema in the AWS impementation.  I can probably get 90% of the way there with an AWS-specific ruleset.

**Comment by andylockran** [comment]
I've got 90% of the way there with an initial attempt at an AWS-specific ruleset:  https://github.com/andylockran/spectral-aws-apigateway-ruleset

**Comment by P0lip** [comment]
Hey!
It's possible in Spectral v6.
You'll only need to use the JS ruleset (`.spectral.js`).
Here's how you can do it:

```js
import { oas } from "@stoplight/spectral-rulesets";
const myFormat = document => typeof document === "object" && document !== null && "x-lint" in document;
const oasRulesetWithOverriddenFormats = {
  ...oas,
  formats: [myFormat],
  rules: Object.entries(oas.rules).reduce((rules, [name, rule]) => {
    rules[name] =
      "formats" in rule
        ? {
            ...rule,
            formats: [myFormat],
          }
        : rule;
    return rules;
  }, {}),
};
export default {
  extends: [oasRulesetWithOverriddenFormats],
};
```

The example is pretty basic, but hopefully, it gives you a grasp of things.
Reach out to me here please, should you have any questions.
